gcloud-java-logging/src/main/java/com/google/cloud/logging/MetricInfo.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.logging.v2.LogMetric;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Google Cloud Logging metrics describe logs-based metric. The value of the metric is the number of
+ * log entries that match a logs filter (see {@link #filter()}).
+ *
+ * @see <a href="https://cloud.google.com/logging/docs/view/logs_based_metrics">Logs-based Metrics
+ *     </a>
+ */
+public class MetricInfo implements Serializable {
+
+  private static final long serialVersionUID = 666208243838820325L;
+
+  private final String name;
+  private final String description;
+  private final String filter;
+
+  /**
+   * A builder for {@code MetricInfo} objects.
+   */
+  public abstract static class Builder {
+
+    /**
+     * Sets the name of the metric. Example: {@code severe-errors}. Metric identifiers are
+     * limited to 1000 characters and can include only the following characters: {@code A-Z},
+     * {@code a-z}, {@code 0-9}, and the special characters {@code _-.,+!*',()%/\}. The
+     * forward-slash character ({@code /}) denotes a hierarchy of name pieces, and it cannot be the
+     * first character of the name.
+     */
+    public abstract Builder name(String name);
+
+    /**
+     * Sets an optional description for this metric. Used for documentation purpose.
+     */
+    public abstract Builder description(String description);
+
+    /**
+     * Sets an advanced logs filter. The value of the metric is the number of log entries that match
+     * this filter. Example: {@code logName=projects/my-projectid/logs/syslog AND severity>=ERROR}.
+     *
```

Comment:
Should there be a shortcut method for specifying just a log("blah") that creates the logname=projects/my-projectid/logs/blah?